x-pack/plugins/apm/server/routes/settings/custom_link/get_transaction.ts

```diff
@@ -48,5 +49,5 @@ export async function getTransaction({
     'get_transaction_for_custom_link',
     params
   );
-  return resp.hits.hits[0]?._source;
+  return resp.hits.hits[0]?._source as Transaction;
```

Comment:
Can you please remove the casting and try running the `checktypes`. Its quite weird that these now need dedicated casting.
I don't see any significant changes here

![image](https://github.com/elastic/kibana/assets/7416358/28a6b8f7-c6ef-4538-af56-1b6b30e55680)

Reply:
Actually, not sure why it's still using the events, probably I changed back for some reason and forget the casting, I will review it again

Reply:
So, now using sources with documentType

![image](https://github.com/elastic/kibana/assets/31922082/97a284d3-383a-4364-abf7-37bbad389e41)

![image](https://github.com/elastic/kibana/assets/31922082/f60d5cce-28d6-4d0b-9e4b-cb2db2130560)